<a href="https://colab.research.google.com/github/EL919177/EL919177/blob/main/Project2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np

For this Project, we will be looking at 2 datasets both related to healthcare.

One is in regards to the health risk factors for Heart Disease, the other is on healthy lifestyles in a city by city basis. 

We will take a look at each of these and talk about each of their particular characteristics. 

In [ ]:
df=pd.read_csv('/content/drive/MyDrive/07 Week 07/heart_2020_cleaned.csv')
df.head()

HeartDisease    BMI Smoking AlcoholDrinking Stroke  PhysicalHealth  \
0           No  16.60     Yes              No     No             3.0   
1           No  20.34      No              No    Yes             0.0   
2           No  26.58     Yes              No     No            20.0   
3           No  24.21      No              No     No             0.0   
4           No  23.71      No              No     No            28.0   

   MentalHealth DiffWalking     Sex  AgeCategory   Race Diabetic  \
0          30.0          No  Female        55-59  White      Yes   
1           0.0          No  Female  80 or older  White       No   
2          30.0          No    Male        65-69  White      Yes   
3           0.0          No  Female        75-79  White       No   
4           0.0         Yes  Female        40-44  White       No   

  PhysicalActivity  GenHealth  SleepTime Asthma KidneyDisease SkinCancer  
0              Yes  Very good        5.0    Yes            No        Yes  
1              Yes  Very good        7.0     No            No         No  
2              Yes       Fair        8.0    Yes            No         No  
3               No       Good        6.0     No            No        Yes  
4              Yes  Very good        8.0     No            No         No

In [ ]:
len(df), len(df.columns)

(319795, 18)

The source of the data for the heart desease dataset is the 2020 annual CDC survey data compiled from 400k adults compiled by the Behavioral Risk Factor Surveillance System (BRFSS) phone surveys. 

The goal of the data set is to determine which risk factors are heavily correlated with increase in heart disease as the target. This is a Classification problem as a result.

There are a total of 319795 rows, 18 columns(17 features), and the issue to look out for are for duplicates as well as figuring the strongest correlating factors to the target column. The Balance issues can be remedied with logistic regression. 

In [ ]:
df=pd.read_csv('/content/drive/MyDrive/07 Week 07/healthy_lifestyle_city_2021.csv')
df.head()

City  Rank Sunshine hours(City) Cost of a bottle of water(City)  \
0   Amsterdam     1                 1858                           £1.92   
1      Sydney     2                 2636                           £1.48   
2      Vienna     3                 1884                           £1.94   
3   Stockholm     4                 1821                           £1.72   
4  Copenhagen     5                 1630                           £2.19   

  Obesity levels(Country)  Life expectancy(years) (Country)  \
0                  20.40%                              81.2   
1                  29.00%                              82.1   
2                  20.10%                              81.0   
3                  20.60%                              81.8   
4                  19.70%                              79.8   

  Pollution(Index score) (City) Annual avg. hours worked  \
0                         30.93                     1434   
1                         26.86                     1712   
2                         17.33                     1501   
3                         19.63                     1452   
4                         21.24                     1380   

   Happiness levels(Country)  Outdoor activities(City)  \
0                       7.44                       422   
1                       7.22                       406   
2                       7.29                       132   
3                       7.35                       129   
4                       7.64                       154   

   Number of take out places(City) Cost of a monthly gym membership(City)  
0                             1048                                 £34.90  
1                             1103                                 £41.66  
2                             1008                                 £25.74  
3                              598                                 £37.31  
4                              523                                 £32.53

In [ ]:
len(df), len(df.columns)

(44, 12)

The source of the data for this dataset is the Healthy Lifestyle Cities Report 2021 compiled from various travel, worksite, and health related sources by Lenstore Vision Care Experts.

The goal of the data set is to determine which Cities have a greater rank in healthiness factor which is determined by Obesity levels, Life Expectancy, and other factors. This is a Regression problem.

There are a total of 44 rows, 12 columns(11 features), and the issue to look out for is modeling the data by several of the features. 